lib/improver/wind_downscaling.py

```diff
@@ -0,0 +1,967 @@
+            yres = (np.diff(a_cube.coord(y_name).bounds)).mean()
+        except Exception:
+            xres = (np.diff(a_cube.coord(x_name).points)).mean()
+            yres = (np.diff(a_cube.coord(y_name).points)).mean()
+        if (
+                (a_cube.coord(x_name).units != exp_unit) or
+                (a_cube.coord(y_name).units != exp_unit)):
+            raise ValueError("cube axis have units different from m.")
+        return (xres + yres) / 2.0
+
+    @staticmethod
+    def check_ancils(a_over_s_cube, sigma_cube, z0_cube, pp_oro_cube,
+                     model_oro_cube):
+        """Check ancils grid and units.
+
+        Check if ancil cubes are on the same grid and if they have the
+        expected units. The testing for "same grid" might be replaced
+        if there is a general utils function made for it or so.
+
+        Parameters:
+        ----------
+        a_over_s_cube: iris cube
+            holding the silhoutte roughness field
+        sigma_cube: iris cube
+            holding the standard deviation of height in a grid cell
+        z0_cube: iris cube or None
+            holding the vegetative roughness field
+        pp_oro_cube: iris cube
+            holding the post processing grid orography
+        model_oro_cube: iris cube
+            holding the model orography on post processing grid
+
+        Returns:
+        -------
+        logical
+            describing whether or not the tests passed
+
+        """
+        alist = [a_over_s_cube, sigma_cube, pp_oro_cube, model_oro_cube]
+        unwanted_coord_list = [
+            "time", "height", "model_level_number", "forecast_time",
+            "forecast_reference_time", "forecast_period"]
+        for field, exp_unit in zip(alist, [None, Unit("m"), Unit("m"),
+                                           Unit("m")]):
+            for unwanted_coord in unwanted_coord_list:
+                try:
+                    field.remove_coord(unwanted_coord)
+                except Exception:
+                    pass
+            if field.units != exp_unit:
+                msg = ('{} ancil field has unexpected unit:'
+                       ' {} (expected) vs. {} (actual)')
+                raise ValueError(
+                    msg.format(field.name(), exp_unit, field.units))
+        if z0_cube is not None:
+            alist.append(z0_cube)
+            for unwanted_coord in unwanted_coord_list:
+                try:
+                    z0_cube.remove_coord(unwanted_coord)
+                except Exception:
+                    pass
+            if z0_cube.units != Unit('m'):
+                msg = ("z0 ancil has unexpected unit: should be {} "
+                       "is {}")
+                raise ValueError(msg.format(Unit('m'), z0_cube.units))
+        mlist = list(itertools.permutations(alist, 2))
+        oklist = []
+        for entr in mlist:
+            oklist.append(entr[0].coords() == entr[1].coords())
```

Comment:
This check is on all coordinates within an ancillary file. Depending upon how the ancillary files will be processed ahead of getting to this point, there may be some coordinates on the cube, which do not match, for example, the time coordinate, depending upon if there is a time coordinate on the cubes. It might be worth restricting this check to only the x and y dimensions, as I think that is all that is required at this point, as mismatches in the time coordinate, may not be relevant for ancillary files.